internal/appsec/appsec.go

```diff
@@ -29,16 +31,40 @@ func Enabled() bool {
 // Start AppSec when enabled is enabled by both using the appsec build tag and
 // setting the environment variable DD_APPSEC_ENABLED to true.
 func Start(opts ...StartOption) {
+	// AppSec can start either:
+	// 1. Manually thanks to DD_APPSEC_ENABLED
+	// 2. Remotely when DD_APPSEC_ENABLED is undefined
+	// Note: DD_APPSEC_ENABLED=false takes precedence over remote configuration
+	// and enforces to have AppSec disabled.
 	enabled, set, err := isEnabled()
 	if err != nil {
 		logUnexpectedStartError(err)
 		return
 	}
+
 	// Check if AppSec is explicitly disabled
 	if set && !enabled {
 		log.Debug("appsec: disabled by the configuration: set the environment variable DD_APPSEC_ENABLED to true to enable it")
 		return
 	}
+
+	// Check whether libddwaf - required for Threats Detection - can be enabled or not
+	if ok, err := waf.Load(); err != nil {
```

Comment:
It's is not a big deal until we remove the appsec build tag but still

Reply:
I don't think there's any value in waiting for a remote activation update, when we can error out at the very start of appsec. Basically what this does is error out on appsec.Start() if the WAF is not useable, instead of erroring out in the RC update that activates appsec in case of remote activation. No sense in "allowing" remote activation if we know it's going to fail.

Reply:
Yes we want to do as most check as possible early on. But do we want to have the WAF loaded (and it's memory footprint) even if appsec will probably never be enabled ?

Reply:
Should be fixed by https://github.com/DataDog/dd-trace-go/pull/2090/commits/4dbdfb0b11df37e4d8174259dd748d09592b6005